### ISSUE TITLE
ChartMPL redraws too readily

### ISSUE THREAD

**Issue by AndrewAnnex** [pyvista/pyvista#2932]
### Describe the bug, what's wrong, and what you expected.

in ChartMPL, a callback to self._redraw is connected to the canvas on `draw_event` https://github.com/pyvista/pyvista/blob/fb108591f941f7452c05905da3fb9c13a54f92dc/pyvista/plotting/charts.py#L4184 which grabs the canvas buffer and makes a new vtkImageData object. 

The problem is that the `draw_event` happens rapidly, even just by moving the mouse across the plot window without any updates to the plot dimensions or underlying plot image or data. This causes performance issues.

Ideally, the _redraw method shouldn't be called so often, or some caching could be used to speed this up. Matplotlib is a very great library to rely upon for plotting but I wouldn't expect the canvas to be redrawn so often. 

### Steps to reproduce the bug.

```python 
 Insert code here 
 ```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sat Jul 02 20:15:06 2022 PDT
                OS : Darwin
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 16.0 GiB
       Environment : IPython
       File system : apfs
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon Pro 560 OpenGL Engine
       GPU Version : 4.1 ATI-4.6.21
  Python 3.10.4 | packaged by conda-forge | (main, Mar 24 2022, 17:43:32)
  [Clang 12.0.1 ]
           pyvista : 0.34.0
               vtk : 9.1.0
             numpy : 1.21.6
           imageio : 2.16.1
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.2
         pyvistaqt : 0.9.0
           IPython : 8.3.0
          colorcet : 3.0.0
        ipyvtklink : 0.2.2
             scipy : 1.8.0
              tqdm : 4.63.0
            meshio : 5.3.4
--------------------------------------------------------------------------------
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dcbr** [comment]
Thanks for the report. I'm sure the performance of the matplotlib chart integration can be further improved, as this was just an initial implementation. However, I'm not sure the cause is the `draw_event` callback. I believe it's quite crucial to have this callback, as otherwise any changes to the matplotlib plot won't be updated in the pyvista Plotter window. We can probably significantly reduce the amount of redraws in the `_render_event` method though, as this seems a bit excessive right now. I'll try to run some experiments to verify this.

Could you also provide some example code? Because when I run following script, I do not see a `_redraw` call when moving the mouse (only when clicking or resizing the Plotter window):

```py
import pyvista as pv

p = pv.Plotter()
chart = pv.ChartMPL()
p.add_chart(chart)

chart._canvas.mpl_connect('draw_event', lambda *args: print("redraw"))

p.show()
```

**Comment by AndrewAnnex** [comment]
@dcbr try clicking and dragging across the plotter window. I am developing a little app for my research which uses a multiplot so the following updated code example should illustrate the issue:

```
import pyvista as pv
   ...:
   ...: p = pv.Plotter(shape=(1,2))
   ...: p.subplot(0, 0)
   ...: p.add_text("Render Window 1", font_size=30)
   ...: p.add_mesh(pv.Cube(), show_edges=True, color="tan")
   ...:
   ...: p.subplot(0,1)
   ...: chart = pv.ChartMPL()
   ...: p.add_chart(chart)
   ...:
   ...: chart._canvas.mpl_connect('draw_event', lambda *args: print("redraw"))
   ...:
   ...: p.show()
```

I agree the call to `_redraw` is needed when the canvas is updated, but currently the redraw method is being called without any significant or real state change. Maybe some timeout based cache could be used or if there is a more specific event to use (like if the canvas data changes or if the window is resized).

**Comment by AndrewAnnex** [comment]
@dcbr just quickly tried it with the code example I had above, I think #2943 solves this issue for me!

**Comment by AndrewAnnex** [comment]
@dcbr, revising my comment, I think it does resolve the performance issue, just to confirm it should redraw the canvas if the underlying plot is updated ('draw_event') is called, but render_event will only trigger if the plotter window size is changed?

**Comment by dcbr** [comment]
Exactly, I verified the dynamic updates still work by running the [chart_overlays example](https://docs.pyvista.org/examples/02-plot/chart_overlays.html).
